pkg/pillar/cmd/nim/nim.go

```diff
@@ -674,6 +661,12 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject) in
 					handleLinkChange(&nimCtx)
 					handleInterfaceChange(&nimCtx, ifindex,
 						"LinkChange", true)
+
+					// If an interface that is part of current DPC has changed (ADD/DELETE),
+					// we should re-start DNS verification to setup DHCP and get addresses if any.
+					if !nimCtx.deviceNetworkContext.Pending.Inprogress {
+						devicenetwork.MaybeStartDhcpClient(log, ifindex, *nimCtx.deviceNetworkContext.DevicePortConfig)
+					}
```

Comment:
See comment below.

Reply:
1. Say eth1 is UP and has dhcpcd running.
2. unbind driver from eth1. Interface eth1 vanishes and so will dhcpcd (terminates).
3. bind driver to eth1 again.
With out this change, we do start dhcpcd. So we will never get IP address on eth1.